frontend/packages/ux-editor/src/components/config/editModal/EditOptions/OptionTabs/EditTab/OptionListEditor/OptionListEditor.tsx

```diff
@@ -7,26 +7,44 @@ import type { SelectionComponentType } from '../../../../../../../types/FormComp
 import { useOptionListQuery } from 'app-shared/hooks/queries';
 import { LibraryOptionsEditor } from './LibraryOptionsEditor';
 import { ManualOptionsEditor } from './ManualOptionsEditor';
+import { handleOptionsChange, resetComponentOptions } from '../../utils/optionsUtils';
 
 export type OptionListEditorProps = Pick<
   IGenericEditComponent<SelectionComponentType>,
   'component' | 'handleComponentChange'
 >;
 
-export function OptionListEditor({
-  component,
-  handleComponentChange,
-}: OptionListEditorProps): React.ReactNode {
+export const OptionListEditor = forwardRef<HTMLDialogElement, OptionListEditorProps>(
+  ({ component, handleComponentChange }: OptionListEditorProps, dialogRef): React.ReactNode => {
+    const handleDelete = () => {
+      const updatedComponent = resetComponentOptions(component);
+      handleOptionsChange(updatedComponent, handleComponentChange);
+    };
+
+    if (component.options !== undefined) {
+      return (
+        <ManualOptionsEditor
+          ref={dialogRef}
+          component={component}
+          handleComponentChange={handleComponentChange}
+          handleDelete={handleDelete}
+        />
+      );
+    }
+
+    return <LibraryEditor component={component} handleDelete={handleDelete} />;
```

Comment:
Nitpicking, but this should be named something like `OptionListResolver` since all it does is waiting for data before rendering the actual `LibraryOptionsEditor`.